connection_helper.go

```diff
@@ -39,3 +42,12 @@ func (c *Connection) writeQuery(token int64, q []byte) error {
 
 	return c.writeData(data)
 }
+
+func (c *Connection) contextFromConnectionOpts() context.Context {
+	sum := c.opts.ReadTimeout + c.opts.WriteTimeout
```

Comment:
Previous behaviour is a quite useless. If server has continuous queries flow, socket's deadline updates every time so that even first query will not timeout never due to this reason. For me it seems like a bug.
And this is unexpected if you don't know that timeouts from config are just passed to socket. 
The new behaviour is that every query ends not more than read+write timeout.